niaaml/pipeline.py

```diff
@@ -509,10 +516,15 @@ def evaluate_pipeline(
 
         classifier.fit(x_train, y_train)
         predictions = classifier.predict(x_test)
+
+        # infer task for the selection of the OptimizationStats
+        # only regression models have the "Task" attribute
+        classification = not hasattr(classifier, 'Task')
+
         return (
             fitness_function.get_fitness(predictions, y_test) * -1,
             selected_features_mask,
-            OptimizationStats(predictions, y_test),
+            OptimizationStats(predictions, y_test) if classification else RegressionOptimizationStats(predictions, y_test),
```

Comment:
Even though a more general `OptimizationStats` class would have been prettier and better to maintain in the future, implementing it this way is less invasive and makes sure that other code depending on this class still works.